src/main/java/com/gregtechceu/gtceu/api/misc/virtualregistry/entries/VirtualRedstone.java

```diff
@@ -0,0 +1,75 @@
+package com.gregtechceu.gtceu.api.misc.virtualregistry.entries;
+
+import com.gregtechceu.gtceu.api.misc.virtualregistry.EntryTypes;
+import com.gregtechceu.gtceu.api.misc.virtualregistry.VirtualEntry;
+
+import net.minecraft.nbt.CompoundTag;
+
+import lombok.Getter;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class VirtualRedstone extends VirtualEntry {
+
+    private static final String MEMBERS_KEY = "members";
+
+    @Getter
+    private final Map<UUID, Short> members = new HashMap<>();
```

Comment:
Use a fastutil `Object2ShortMap`

```suggestion
    private final Object2ShortMap<UUID> members = new Object2ShortOpenHashMap<>();
```